src/dune_pkg/opam_solver.ml

```diff
@@ -424,27 +413,34 @@ let opam_package_to_lock_file_pkg ~repo ~local_packages opam_package =
       Loc.none, Package_name.of_string (OpamPackage.Name.to_string name))
   in
   let build_command =
+    let subst_step =
+      OpamFile.OPAM.substs opam_file
+      |> List.map ~f:(fun x ->
+        let x = OpamFilename.Base.to_string x in
+        let input = String_with_vars.make_text Loc.none (x ^ ".in") in
+        let output = String_with_vars.make_text Loc.none x in
+        Action.Substitute (input, output))
+    in
     let patch_step =
       (* CR-someday alizter: Patches don't take into account filters that are present. For
          now we take them all. *)
-      match
-        OpamFile.OPAM.patches opam_file
-        |> List.map ~f:(fun (x, _) ->
-          Action.Patch
-            (String_with_vars.make_text Loc.none (OpamFilename.Base.to_string x)))
-      with
-      | [] -> None
-      | [ x ] -> Some x
-      | xs -> Some (Action.Progn xs)
+      OpamFile.OPAM.patches opam_file
+      |> List.map ~f:(fun (x, _) ->
+        Action.Patch (String_with_vars.make_text Loc.none (OpamFilename.Base.to_string x)))
```

Comment:
FYI `Action.When` should work here to apply the filter (happy for this to come in a later change though)

Reply:
Let's do that in a later change. Thanks!